unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/nio/channels/DatagramChannelTest.scala

```diff
@@ -0,0 +1,139 @@
+package org.scalanative.testsuite.javalib.nio.channels
+
+import org.junit.Test
+import org.junit.Assert._
+import org.junit.Assume._
+import org.junit.BeforeClass
+import org.junit.Ignore
+
+import org.scalanative.testsuite.utils.AssertThrows.assertThrows
+import org.scalanative.testsuite.utils.Platform
+
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.nio.channels._
+import java.net.{InetAddress, InetSocketAddress}
+
+object DatagramChannelTest {
+  private lazy val loopback = InetAddress.getLoopbackAddress()
+}
+
+class DatagramChannelTest {
+  import DatagramChannelTest._
+
+  private def withDatagramChannel(f: DatagramChannel => Unit): Unit = {
+    val dc = DatagramChannel.open()
+    try {
+      f(dc)
+    } finally {
```

Comment:
Nice to see channel getting closed on both success and exception paths.  We have a _lot_ of technical debt in SN with this not happening.